app/lib/katello/validators/content_view_environment_coherent_default_validator.rb

```diff
@@ -0,0 +1,22 @@
+module Katello
+  module Validators
+    # ensures that the Default Organization View content view can only be used with the Library environment
+    class ContentViewEnvironmentCoherentDefaultValidator < ActiveModel::Validator
+      def validate(record)
+        #support lifecycle_environment_id for foreman models
+        environment_id = record.respond_to?(:lifecycle_environment_id) ? record.lifecycle_environment_id : record.environment_id
+        if record.content_view_id
```

Comment:
... I suppose it doesn't really matter because this validator is used only for content view environments, and those should always have an environment. I guess in that  case the `respond_to?` check is a bit irrelevant, but again, maybe there is some benefit in keeping it exactly the same as the other related validator.

Reply:
Okay, took me a minute to remember, but I remembered the reason :smile: 

It's about making sure we raise the correct errors.

Content view environments already have these validations:
```rb
validates :environment_id, uniqueness: {scope: :content_view_id}, presence: true
    validates :content_view_id, presence: true
```

Omitting one of them would normally cause an error like `environment_id must be specified`, because of the `presence: true` validation.

But let's say these checks aren't there. You'd get _both_ the correct and incorrect error:

```
environment_id must be specified, Lifecycle environment '' cannot be used with content view 'myCV'
```

Reply:
makes sense!